src/Stripe.net/Infrastructure/StringUtils.cs

```diff
@@ -52,5 +55,25 @@ public static bool SecureEquals(string a, string b)
 
             return result == 0;
         }
+
+        /// <summary>
+        /// Validates that the API key doesn't contain any whitespace characters.
+        /// </summary>
+        /// <param name="apiKey">The API key.</param>
+        /// <returns>The API key, if valid.</returns>
+        /// <exception name="StripeException">
+        /// Thrown if the API key contains whitespace characters.
+        /// </exception>
+        public static string ValidateApiKey(string apiKey)
+        {
+            if (!apiKeyRegex.IsMatch(apiKey ?? string.Empty))
+            {
+                var message = $"Invalid API key: \"{apiKey}\". API keys may not contain whitespace "
+                    + "characters.";
+                throw new StripeException(message);
```

Comment:
Minor but do we really want to output the API key here?

Reply:
I thought it would help in figuring out the issue, e.g.:
```
Invalid API key: "sk_test_123
". API keys may not contain whitespace characters.
```
The line break before the closing quote helps identify the invalid `\n` character.

Reply:
Agreed but I wonder if it makes people leak their API key if they are not careful. That seems quite dangerous to do that right? I know it's a major revision, but still.